tests/pytest_framework/functional_tests/destination_drivers/file_destination/test_file_reopen.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2015-2018 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import pathlib2
+
+class FileReopenTest(object):
+    def __init__(self, tc):
+        self.config = tc.new_config()
+        self.file_source = self.config.create_file_source(file_name="input.log")
+        source_group = self.config.create_source_group(self.file_source)
+
+        self.file_destination = self.config.create_file_destination(file_name="output.log")
+        self.path = self.file_destination.getPath()
+        destination_group = self.config.create_destination_group(self.file_destination)
+
+        self.new_file_destination = self.config.create_file_destination(file_name="output.old")
+
+        self.config.create_logpath(sources=source_group, destinations=destination_group)
+
+        self.bsd_log = tc.format_as_bsd(tc.create_dummy_bsd_message())
+
+        self.syslog_ng = tc.new_syslog_ng()
+
+    def start(self):
+        self.syslog_ng.start(self.config)
+
+    def send_message(self):
+        self.file_source.write_log(self.bsd_log)
+
+    def rename_destination(self):
+        self.path.rename(pathlib2.Path(str(self.path.parent), self.path.stem + ".old"))
```

Comment:
rename also accepts strings, IMHO it will be easier to read this line without the extra constructor.

Reply:
I used the constructor to properly concatenate the directory part and the filename part. I could use os.path.join too, but that would introduce the same complexity.

Reply:
if you imported Path into the local module (e.g. from pathlib2 import Path), the result would be more readable. Also, can't you get rid off the extra str() call? That would turn this into:

```
self.path.rename(Path(self.path.parent, self.path.stem + ".old"))
```